### ISSUE TITLE
unsupported-assignment-operation false positive

### ISSUE THREAD

**Issue by nikita-leontiev** [pylint-dev/pylint#10050]
### Bug description

Below code leads to unsupported-assignment-operation false positive:
```
if __name__ == "__main__":
    test_dict = {"key": None}
    test_dict["key"] = {"key": None}
    test_dict["key"]["key"] = 0
```

### Configuration

Without pylintrc.

### Command used

pylint -sn --disable=C0114 test.py

### Pylint output

```
test.py:4:4: E1137: "test_dict['key']" does not support item assignment (unsupported-assignment-operation)
```

### Expected behavior

No E1137.

### Pylint version

pylint 3.2.7
astroid 3.2.4
Python 3.8.10 (default, Sep 11 2024, 16:02:53)
[GCC 9.4.0]

### OS / Environment

Ubuntu 20.04
